src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java

```diff
@@ -38,7 +38,7 @@
 	public Link(String id, String anchor, int start, int end) {
 		super();
 		try {
-			this.id = java.net.URLDecoder.decode(id, "UTF-8");
+			this.id = java.net.URLDecoder.decode(id.replace("+", "%2B"), "UTF-8");
```

Comment:
Trying to decode is not a hack.

This is due to how we expected decode to behave, on java: https://docs.oracle.com/javase/7/docs/api/java/net/URLDecoder.html (`+` get transformed into spaces.)
we assumed it was like the python urllibs' implementation which would leave anything htat is not urlencoded untouched.